### PR TITLE
feat(third-party): bump `gateway-api` to v1.0.0

### PR DESCRIPTION
**Pull request by leozhantw**
## Desc
Bump `gateway-api` to [v1.0.0](https://github.com/kubernetes-sigs/gateway-api/releases/tag/v1.0.0).

### CODE REVIEW

**Review by tommy351 — COMMENTED** [review]
You have to create a changeset. Please follow [the guide](https://github.com/tommy351/kubernetes-models-ts/blob/master/CONTRIBUTING.md#changeset) here.

**Comment by leozhantw** [comment]
> You have to create a changeset. Please follow [the guide](https://github.com/tommy351/kubernetes-models-ts/blob/master/CONTRIBUTING.md?rgh-link-date=2023-11-23T03%3A14%3A43Z#changeset) here.

Fixed it, thx.

File: third-party/gateway-api/package.json [hunk 1]
```diff
@@ -43,7 +43,7 @@
   },
   "crd-generate": {
     "input": [
-      "https://github.com/kubernetes-sigs/gateway-api/releases/download/v0.6.2/standard-install.yaml"
+      "https://github.com/kubernetes-sigs/gateway-api/releases/download/v1.0.0/standard-install.yaml"
```
[the remark this answers is not in the document]
Comment (leozhantw) [reply third-party/gateway-api/package.json:47]: Thx!

Adjusted it.

File: .changeset/tiny-berries-sneeze.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@kubernetes-models/gateway-api": patch
```
Comment (tommy351) [inline .changeset/tiny-berries-sneeze.md:2]:
```suggestion
"@kubernetes-models/gateway-api": minor
```

File: third-party/gateway-api/__tests__/class.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import { describe, it, expect, beforeEach } from "vitest";
-import { HTTPRoute } from "../gen/gateway.networking.k8s.io/v1beta1/HTTPRoute";
+import { HTTPRoute  } from "../gen/gateway.networking.k8s.io/v1/HTTPRoute";
```
Comment (tommy351) [inline third-party/gateway-api/__tests__/class.ts:2]:
```suggestion
import { HTTPRoute } from "../gen/gateway.networking.k8s.io/v1/HTTPRoute";
```

Comment (leozhantw) [reply third-party/gateway-api/__tests__/class.ts]: Fixed. 🙇🏻

File: .changeset/tiny-berries-sneeze.md (hunk 2 above)
Comment (leozhantw) [reply .changeset/tiny-berries-sneeze.md]: Adjusted it 🙏🏻

**Pull request merged**